libcodechecker/libhandlers/server.py

```diff
@@ -342,6 +369,193 @@ def arg_match(options):
     parser.set_defaults(func=__handle)
 
 
+def handle_connect_ret(ret):
+    """
+    Print out a message based on the database statuses.
+    """
+    if ret == DBStatus.OK:
+        return "Database is OK."
+    elif ret == DBStatus.FAILED_TO_CONNECT:
+        return "Failed to connect to the configured database"
+    elif ret == DBStatus.SCHEMA_INITIALIZATION_ERROR:
+        return "Schema initialization error"
+    elif ret == DBStatus.SCHEMA_MISSING:
+        return "Schema is missing"
+    elif ret == DBStatus.SCHEMA_MISMATCH_OK_MIGRATION:
+        return "Schema mismatch"
+    elif ret == DBStatus.SCHEMA_MISMATCH_NO_MIGRATION:
+        return "Automatic migration is not available"
+    elif ret == DBStatus.MISSING:
+        return "Database is missing"
+
+
+def get_user_input(msg):
+    """
+    Get the user input.
+
+    :returns: True/False based on the asnwer from the user
+    """
+    yes = ['y', 'Y', 'yes', 'Yes']
+
+    answer = raw_input(msg)
+    if answer in yes:
+        return True
+    return False
+
+
+def print_prod_status(prod_status):
+    """
+    Print the database statuses for each of the products.
+    """
+
+    header = ['Product endpoint', 'Database status']
+    rows = []
+
+    for k, v in prod_status.items():
+        rows.append([k, handle_connect_ret(v)])
+
+    prod_status = output_formatters.twodim_to_str('table',
+                                                  header,
+                                                  rows,
+                                                  sortby_column_number=0)
+    print(prod_status)
+
+
+def check_product_db_status(cfg_sql_server, context):
+    """
+    Check the products for database statuses.
+
+    :returns: dictionary of product endpoints with database statuses
+    """
+
+    LOG.debug("Checking config database")
+    migration_root = context.run_migration_root
+
+    engine = cfg_sql_server.create_engine()
+    config_session = sessionmaker(bind=engine)
+    sess = config_session()
+
+    products = sess.query(ORMProduct).all()
+
+    prod_status = {}
+    for pd in products:
+        db = database.SQLServer.from_connection_string(pd.connection,
+                                                       RUN_META,
+                                                       migration_root,
+                                                       interactive=False)
+
+        ret = db.connect()
+        prod_status[pd.endpoint] = ret
+
+    sess.commit()
+    sess.close()
+    engine.dispose()
+
+    return prod_status
+
+
+def __db_status_check(cfg_sql_server, context, product_name=None):
+    """
+    Check and print database statuses for the given product.
+    """
+    if not product_name:
+        return 0
+
+    print(product_name)
```

Comment:
Same as above. Use logger instead of print.

Reply:
Using debug log would mess up the generated table output here.